src/core/ledger/grainIntegration.js

```diff
@@ -0,0 +1,127 @@
+// @flow
+
+import {Ledger, type PayoutAddress, type CurrencyId} from "./ledger";
+import {type Distribution} from "./distribution";
+import {getDistributionBalances} from "./distributionSummary/distributionSummary.js";
+import {type Currency} from "./currency.js";
+import {type IdentityId} from "../identity";
+import * as NullUtil from "../../util/null";
+import * as G from "./grain";
+
+type Transfer = {|
+  amount: G.Grain,
+  memo: string,
+|};
+
+export type PayoutDistributions = Array<[PayoutAddress, G.Grain]>;
+export type PayoutAddressToId = Map<PayoutAddress, IdentityId>;
+export type TransferredGrain = Array<[PayoutAddress, Transfer]>;
```

Comment:
I expected you to add payoutAddress as a field to the Transfer type. Why not do that instead? It would make Transfer semantically more whole and simplify derived types.

(also TransferredGrain -> Transfers)